### PR TITLE
[MIG][11.0] hr_attendance_autoclose

### PR DESCRIPTION
**Pull request by jarroyomorales**
Migration of hr_attendance_autoclose to 11.0

### CODE REVIEW

File: hr_attendance_autoclose/data/hr_attendance.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.cron" id="check_attendance_cron">
+        <field name="name">Check Attendance</field>
+        <field name="interval_number">1</field>
+        <field name="interval_type">hour</field>
+        <field name="numbercall">-1</field>
+        <field name="nextcall" eval="(DateTime.now() + timedelta(hour=1)).strftime('%Y-%m-%d %H:05:00')"/>
```
Comment (jarroyomorales) [inline hr_attendance_autoclose/data/hr_attendance.xml]: I am having problems with this. The field seems to be computing properly but after some testing it looks like the cron is ignoring this value and executing 1 hour after the last one finished and not at 10:05, 11:05, 12:05 as it should. I don't know if this trick has stopped working on 11.0 or I am doing something wrong.

File: hr_attendance_autoclose/data/hr_attendance.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.cron" id="check_attendance_cron">
+        <field name="name">Check Attendance</field>
+        <field name="interval_number">1</field>
+        <field name="interval_type">hours</field>
+        <field name="numbercall">-1</field>
+        <field name="nextcall" eval="(DateTime.now() + timedelta(minutes=60)).strftime('%Y-%m-%d %H:05:00')"/>
```
Comment (jarroyomorales) [inline hr_attendance_autoclose/data/hr_attendance.xml:8]: I am having problems with this. The field seems to be computing properly but after some testing it looks like the cron is ignoring this value and executing 1 hour after the last one finished and not at 10:05, 11:05, 12:05 as it should. I don't know if this trick has stopped working on 11.0 or I am doing something wrong.

Comment (AaronHForgeFlow) [reply hr_attendance_autoclose/data/hr_attendance.xml:8]: I think this is standard Odoo. The next execution time is calculated when the cron is finished, instead the previous execution time. Hope they change that in future versions.

File: hr_attendance_autoclose/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    'name': 'HR Attendance Auto Close',
+    'version': '11.0.1.0.0',
+    'category': 'Human Resources',
+    'summary': 'Close stale Attendances',
+    'website': 'https://github.com/OCA/hr',
+    'author': 'Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'installable': True,
+    'depends': [
+        'hr_attendance',
```
Comment (luistorresm) [inline hr_attendance_autoclose/__manifest__.py]:
```suggestion
```

Its already on https://github.com/OCA/hr/blob/11.0/hr_attendance_reason/__manifest__.py#L15

File: hr_attendance_autoclose/models/hr_attendance.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    @api.constrains('check_in', 'check_out', 'employee_id')
+    def _check_validity(self):
+        """ If this is an automatic checkout the constraint is invalid
+        as there may be old attendances not closed
+        """
+        reason = self.env['hr.attendance.reason'].search(
+            [('code', '=', 'S-CO')], limit=1)
+        if not reason:
+            return super(HrAttendance, self)._check_validity()
+        for attendance in self:
```
Comment (luistorresm) [inline hr_attendance_autoclose/models/hr_attendance.py]:
```suggestion
        for attendance in self:
        if self.filtered(lambda att: att. attendance_reason_ids and reason in attendance.attendance_reason_ids):
            return True
```

File: hr_attendance_autoclose/__manifest__.py (hunk 3 above)
Comment (jarroyomorales) [reply hr_attendance_autoclose/__manifest__.py]: True, I just changed it.

File: hr_attendance_autoclose/models/hr_attendance.py (hunk 4 above)
Comment (jarroyomorales) [reply hr_attendance_autoclose/models/hr_attendance.py]: Sorry, I didnt understand this change, could you check what I just commited to see if this is what you meant?

Comment (luistorresm) [reply hr_attendance_autoclose/models/hr_attendance.py]: 👍 

Yes @jarroyomorales

That is to avoid the `for`, sentence

Comment (jarroyomorales) [reply hr_attendance_autoclose/models/hr_attendance.py]: :+1:

File: hr_attendance_autoclose/data/hr_attendance.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (nikul-serpentcs) [inline hr_attendance_autoclose/data/hr_attendance.xml]: @jarroyomorales  `<odoo noupdate="1">`

File: hr_attendance_autoclose/models/hr_attendance.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            else:
+                delta = datetime.now() - datetime.strptime(
+                    attendance.check_in, DEFAULT_SERVER_DATETIME_FORMAT)
+                open_worked_hours = delta.total_seconds() / 3600.0
+            attendance.open_worked_hours = open_worked_hours
+        return True
+
+    open_worked_hours = fields.Float(
+        string='Worked hours', compute='_compute_open_worked_hours',
+        store=False)
```
Comment (nikul-serpentcs) [inline hr_attendance_autoclose/models/hr_attendance.py]: @jarroyomorales no need to define `store=False`, compute already store=False

**Review by nikul-serpentcs — COMMENTED** [review]
nitpicking

**Comment by jarroyomorales** [comment]
@nikul-serpentcs Changes attended

File: hr_attendance_autoclose/data/hr_attendance_reason.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
```
Comment (nikul-serpentcs) [inline hr_attendance_autoclose/data/hr_attendance_reason.xml]: same here @jarroyomorales

Comment (jarroyomorales) [reply hr_attendance_autoclose/data/hr_attendance_reason.xml]: :+1:

File: hr_attendance_autoclose/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    'name': 'HR Attendance Auto Close',
+    'version': '11.0.1.0.0',
+    'category': 'Human Resources',
+    'summary': 'Close stale Attendances',
+    'website': 'https://github.com/OCA/hr',
+    'author': 'Odoo Community Association (OCA)',
```
Comment (nikul-serpentcs) [inline hr_attendance_autoclose/__manifest__.py:10]: @aheficent can you please check author name it's right `'author': 'Eficent, Odoo Community Association (OCA)',` or keep as it is.

Comment (AaronHForgeFlow) [reply hr_attendance_autoclose/__manifest__.py:10]: I think it is fine as it is as the copyright in the files. Thanks.

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by nikul-serpentcs — APPROVED** [review]
Only Code Review
LGTM :+1:

**Pull request merged**